### PR TITLE
Add zsh completion for 'docker daemon --runtimes' and 'docker run --r…

### PR DESCRIPTION
**Pull request by sdurrheimer**
Ref. #22983

@vincentbernat 

Signed-off-by: Steve Durrheimer s.durrheimer@gmail.com

### CODE REVIEW

File: contrib/completion/zsh/_docker [hunk 1]
```diff
@@ -299,6 +299,17 @@ __docker_complete_pid() {
...
+__docker_complete_runtimes() {
+    [[ $PREFIX = -*  ]] && return 1
+    integer ret=1
+
+	emulate -L zsh
+    setopt extendedglob
+    local -a runtimes_opts
+    runtimes_opts=(${(ps: :)${(f)${${"$(_call_program commands docker $docker_options info)"##*$'\n'Runtimes: }%%$'\n'^ *}}})
+	_describe -t runtimes-opts "runtimes options" runtimes_opts && ret=0
+}
```
Comment (vincentbernat) [inline contrib/completion/zsh/_docker]: Nitpick: it seems there is a mix of tab and spaces

**Comment by vincentbernat** [comment]
LGTM

File: contrib/completion/zsh/_docker (hunk 1 above)
Comment (sdurrheimer) [inline contrib/completion/zsh/_docker]: fixed

**Comment by vdemeester** [comment]
LGTM 🐯

**Comment by runcom** [comment]
LGTM

**Pull request merged**